tsconfig.eslint.json

```diff
@@ -0,0 +1,4 @@
+{
+    "extends": "./tsconfig.json",
+    "include": ["."]
```

Comment:
If there's no way to avoid this thing, maybe be more specific as in `["PR-Trigger", "*.js"]`?

(And this shows why I'm puzzled at this: its only the js files that are not covered by the existing tsconfig...)

Reply:
My thinking was if `@typescript-eslint/parser` requires a `tsconfig.json` for every file, then `"include": ["."]` makes that explicit. Is there an advantage I'm overlooking to being specific?

Reply:
Specificity is not what I'm concerned about.  I just dislike having
another `tsconfig` just for the sake of the linter.

With your recent change it looks closer to not needing, no?  I tried
this:

* drop `tsconfig.eslint.json` from `.eslintrc.json`
* drop the `// @ts-check` at the top of `PR-Trigger/index.js` (which is
  bad, but it doesn't check anyway; should be done in the future, and
  even better, switch it to ts)
* in `tsconfig.json` make it include `["src", "PR-Trigger", "*.js"]`
* also turn on `allowJs` there

With that, it looks like it builds and lints fine.  Anything I'm missing
in this?

Reply:
I'm in favor :+1: I think TS is limited to one `rootDir` per `tsconfig.json` which slightly rearranges the `bin` dir. I've updated the referencing paths accordingly. I think `function.json` will need `"scriptFile": "../bin/PR-Trigger/index.js"` (output file) but we'll need that anyway if in future we switch it to TS.

> * drop the `// @ts-check` at the top of `PR-Trigger/index.js` (which is
>   bad, but it doesn't check anyway; should be done in the future, and
>   even better, switch it to ts)

Instead of dropping `@ts-check` I added `@ts-expect-error` for the few outstanding errors, so that it does check.

Reply:
Sorry, I didn't think to check how the outputs look like yesterday.  Changing the paths seems too risky to be worth it, so I converted the whole thing to use reference projects, and then I continued with converting the `PR-Trigger` thing to TS too.  Obviously this has lots of conflicts with this PR -- I'd offer to rebase it, but I'm likely to mess up the eslint details.  Hopefully this will make it easier since everything is now in TS.